picard/formats/vorbis.py

```diff
@@ -316,10 +316,16 @@ def _remove_deleted_tags(self, metadata, tags):
             real_name = self._get_tag_name(tag)
             if real_name and real_name in tags:
                 if real_name in ('performer', 'comment'):
-                    tag_type = r"\(%s\)" % tag.split(':', 1)[1]
-                    for item in tags.get(real_name):
-                        if re.search(tag_type, item):
-                            tags.get(real_name).remove(item)
+                    parts = tag.split(':', 1)
+                    if len(parts) == 2:
+                        tag_type_regex = r"\(%s\)$" % parts[1]
+                    else:
+                        tag_type_regex = r"[^)]$"
+                    existing_tags = tags.get(real_name)
+                    for item in existing_tags:
+                        if re.search(tag_type_regex, item):
+                            existing_tags.remove(item)
+                    tags[real_name] = existing_tags
```

Comment:
The only difference I see between this code and the one in apev2.py is `tags.get(real_name)` and `tags.get(real_name, [])`.
Is this intentional? Can we reduce code redundancy somehow?

Reply:
Yes, this is intentional. I guess we could also write `tags.get(real_name, [])` for Vorbis, but it is not needed there.

We could somehow move those 10 lines into a separate function, but this is really one of those cases where I would keep the code separate. The tag handling is very format specific. The interfaces are similar, but not identical and I am not fully convinced that this is generic enough that we can safely assume the code will be identical for all cases.

Reply:
Ok, thanks for the explanation.